CONTRIBUTING.md

```diff
@@ -1,123 +1,126 @@
-2. If you get an issue assigned to you, please post updates at least once a week. It is also preferred for you to send a draft PR as early as you have something working, before it is ready.
+Some of us use VIM with [rust.vim](https://github.com/rust-lang/rust.vim) and
+[rusty-tags](https://github.com/dan-t/rusty-tags). It has fewer features than
+CLion or VSCode, but overall provides a usable setting.
 
-### Submitting the PR
+Refer to [this
+document](https://docs.nearprotocol.com/docs/contribution/nearcore) for details
+on setting up your environment.
 
-Once your change is ready, prepare the PR. The PR can contain any number of commits, but when it is merged, they will all get squashed. The commit names and descriptions can be arbitrary, but the name and the description of the PR must follow the following template:
+# Build Process
 
-```
-<type>: <name>
+Nearcore is a reasonably standard Rust project, so `cargo test` most likely will
+just work.  There are couple of specifics though:
+
+* `nearcore` assumes a UNIX-like environment.  Linux is actively supported; Mac
+is generally supported (the newer M1 chipset may not be well supported); and
+Windows is not supported.
+* `nearcore` build process includes generating bindings to RocksDB via
+`bindgen`, which requires `libclang`.  See [`bindgen`
+documentation](https://rust-lang.github.io/rust-bindgen/requirements.html#clang)
+for installation instructions.
+* At the moment, `nearcore` only supports compiling for x86_64 architecture (for
+technical reasons, we have to care about CPU architecture and supporting one is
+easier).  To compile to x86_64 on Apple silicon, use `rustup set default-host
+x86_64-apple-darwin` command.  That is, cross-compilation sadly doesn't work at
+the time of writing.
+* You can optionally use the system installations of `librocksdb`, `libsnappy`
+and `lz4` in order to speed up the compilation and reduce the build memory
+requirements by setting the `ROCKSDB_LIB_DIR`, `SNAPPY_LIB_DIR` and
+`LZ4_LIB_DIR` environment variables.  These environment variables should point at
+the directory containing the dynamic shared objects (`.so`s or `.dylib`s.)
+
+If your setup does not involve `rustup`, the required version of the rust
+toolchain can be found in the `rust-toolchain` file.
+
+# Pull Requests
+
+All the contributions to `nearcore` happen via Pull Requests.  To create a Pull
+Request, fork `nearcore`, create a new branch, do the work there, and then send
+a PR against the `master` branch.
+
+When working on a PR, please keep the following in mind:
+
+1. The changes should be thoroughly tested.  Please refer to [this
+document](https://github.com/nearprotocol/nearcore/wiki/Writing-tests-for-nearcore)
+for our testing guidelines and overview of the testing infrastructure.
+2. If you get an issue assigned to you, please post updates at least once a
+week.  You are encouraged to submit work-in-progress draft PRs to get early
+feedback.
```

Comment:
I suggest cutting this. We are not assigning people to issues/PRs consistenly, and we certainly do let PRs to linger for more than a week. 

To have this rule written down, we need someone to actually enforce it.

Reply:
Agreed, now removed.